### DISCUSSION TITLE
What is the meaning of `MOA` in the class `MOADNSParser`?

### DISCUSSION THREAD

**Discussion by Chinese-Coding** [PowerDNS/pdns#14989] [Q&A] [answered]
# What is the meaning of `MOA` in the class `MOADNSParser`?
I'm reading the source code of pdns and I am confused about the naming of some of the classes.
Could any one help me answer this question, thank you.

**Comment by paddg** [comment, marked as answer]
"Mother of all" afaik

**Reply by paddg** [reply]
Found also the [reference](https://doc.powerdns.com/recursor/appendices/internals.html#moadnsparser)

**Reply by Chinese-Coding** [reply]
![image](https://github.com/user-attachments/assets/a336e5ed-14f4-4238-a111-89b29f43b995)
That is true! Thank you!
